File: 0118/solution.ts
function generate(numRows: number): number[][] {
  const rows: number[][] = [[1]];
  for (let i = 1; i < numRows; i++) {
    const newRow: number[] = [1];
    for (let j = 0; j < rows[i - 1].length - 1; j++) {
      newRow.push(rows[i - 1][j] + rows[i - 1][j + 1]);
    }
    newRow.push(1);
    rows.push(newRow);
  }
  return rows;
}

File: 0118/solution.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        res.add(Arrays.asList(1));
        for(int i = 2; i <= numRows; i++) {
            List<Integer> curr = new ArrayList<>();
            curr.add(1);
            List<Integer> prev = res.get(res.size() - 1);
            for (int j = 0; j < prev.size() - 1; j++) {
                curr.add(prev.get(j) + prev.get(j + 1));
            }
            curr.add(1);
            res.add(curr);
        }
        return res;
    }
}